src/transformer.js

```diff
@@ -24,6 +24,33 @@ const matcher = (prop, config, TYPE) => {
   }
 }
 
+/**
+ * @typedef transformer
+ * @type Object
+ * @property {Function} map - executes the transformer and returns a new object
+ */
+
+/**
+ * @typedef transformerConfig
+ * @type {Object}
+ * @property {Function} transform - the default transform to use
+ * @property {Array<matcher>} matchers - collection of matcher objects
+ */
+
+/**
+ * @typedef matcher
+ * @type Object
+ * @property {Function} [transform] - function to compute new value from any
+ * value matched by the matcher. If not defined the default transform
+ * will be used.
+ * @property {Rejex | Function} [key] - used to peform matching on keys.
```

Comment:
Is it `Rejex` or `RegExp `? (https://stackoverflow.com/questions/10853582/regexp-type-for-closure-compiler)